core/src/main/java/tachyon/worker/eviction/EvictLRU.java

```diff
@@ -56,18 +56,16 @@ public EvictLRU(boolean lastTier) {
       }
       long blockId = candidate.getSecond();
       long blockSize = dir.getBlockSize(blockId);
-      // Add info of the block to the list
-      blockInfoList.add(new BlockInfo(dir, blockId, blockSize));
-      dir2BlocksToEvict.put(dir, blockId);
-      dir2LRUBlocks.remove(dir);
-      long evictBytes;
-      // Update eviction size for this StorageDir
-      if (sizeToEvict.containsKey(dir)) {
-        evictBytes = sizeToEvict.get(dir) + blockSize;
-      } else {
-        evictBytes = blockSize;
+      long evictBytes = sizeToEvict.containsKey(dir) ? sizeToEvict.get(dir) : 0L;
+      if (blockSize > 0) {
```

Comment:
Blocksize shouldn't be 0, but should we account for it just in case? Since only -1 means block does not exist.